src/main/java/entities/Bucket.java

```diff
@@ -1,49 +1,60 @@
 package entities;
 
+import javax.persistence.*;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.Date;
 import java.util.Objects;
 
+@Entity
+@Table(name = "buckets")
 public class Bucket {
-    private Integer id;
-    private Integer user_id;
-    private Integer product_id;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+    private int user_id;
+    private int product_id;
```

Comment:
It is not by java naming convention to use such names, better to be `userId` and `productId`.